.circleci/config.yml

```diff
@@ -0,0 +1,123 @@
+# These environment variables must be set in CircleCI UI
+#
+# DOCKERHUB_REPO - docker hub repo, format: <username>/<repo>
+# DOCKER_USER
+# DOCKER_PASS
+#
+
+version: 2
+jobs:
+  build:
+    docker:
+      - image: docker:18.02.0-ce
+    working_directory: ~/mozilla/redash
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: |
+          printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' "$CIRCLE_SHA1" "$CIRCLE_TAG" "$CIRCLE_PROJECT_USERNAME" "$CIRCLE_PROJECT_REPONAME" "$CIRCLE_BUILD_URL" > version.json
+      - run:
+          command: docker build -t app:build .
+          no_output_timeout: 20m
+
+  test:
+    docker:
+      - image: circleci/node:6.14.3-stretch
+      - image: redis
+      - image: circleci/postgres:10-alpine-postgis
+    working_directory: ~/mozilla/redash
+    steps:
+      - checkout
+      - run: mkdir -p /tmp/test-reports/pytest
+      - run: sudo apt-get update
+      - run: sudo apt-get install -y python-pip python-dev
+      - run: sudo apt-get install -y redis-tools redis-server
+      - run: sudo pip install --upgrade setuptools
+      - run: sudo pip install -r requirements_dev.txt
+      - run: sudo pip install -r requirements.txt
+      - run: sudo npm install
+      - run: sudo npm run build
+      - run: 
+          command: pytest --junitxml=/tmp/test-reports/pytest/junit.xml tests/
+          environment:
+            REDASH_REDIS_URL: redis://localhost:6379/0
+            REDASH_DATABASE_URL: "postgresql://postgres@localhost/postgres"
+      - store_artifacts: 
+          path: /tmp/test-reports/
+          destination: tr1
+      - store_test_results:
+          path: /tmp/test-reports/
+
+  deploy:
+    docker:
+      - image: docker:18.02.0-ce
+    working_directory: ~/mozilla/redash
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Deploy to Dockerhub
+          no_output_timeout: 20m
+          command: |
+            # Deploy master
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+              ./bin/deploy "master"
+            else
+            # Deploy a release tag...
+              ./bin/deploy "rc"
+            fi
```

Comment:
Could we split the deploy job into two jobs deploy-master and deploy-rc (and update the workflows below) instead of having the shell script with the condition? That would make it more obvious in the Circle UI what's happening.

Reply:
Thanks!